operator/config/samples/component-integration-installed/config_map_istio_2.0.yaml

```diff
@@ -0,0 +1,19 @@
+kind: ConfigMap
+apiVersion: v1
+
+metadata:
+  name: kyma-component-istio
+  namespace: default
+  labels:
+    "operator.kyma-project.io/managed-by": "kyma-operator"
+    "operator.kyma-project.io/controller-name": "istio"
+    "operator.kyma-project.io/release": "v2.0.2"
+data:
+  istio: |-
+    kind: IstioConfiguration
+    group: kyma.kyma-project.io
+    resource: istioconfigurations
+    version: v1alpha1 
+    spec:
+      foo: "created From Dynamic Client for {{installOperation .}} to {{installTargetVersion .}}"
```

Comment:
In the kyma controller, you use labels for `applied-as` and `release`. I think we should stick to labels only and not try to render custom resources with templating. Such templates introduce coupling and are hard to trace/refactor when you rename some variables.

Reply:
Had to test the upgrade somehow :D 

Can remove this without issues, question however: How we are going to inject values in the spec later if not through a template? 

I cannot imagine there will be always a (purely) static configuration for these config maps. Especially for dynamic generation of CRs, I think we need a way to adapt values and create the chart based on some data on the environment or landscape. If its not templating, we should think of something else that fits. We could also think of doing that simply in the component reconciler, but then we limit the CR definitions here to be purely static.

Towards your concern: I think Labels are even harder to trace than Spec changes as they do not introduce a generational change in the status.

Let me know if you still think we should remove it, and we can proceed accordingly

Reply:
{{installTargetVersion .}} = release label from versioned config map
{{installOperation .}} = upgrade/downgrade
{{installOriginVersion .}}  = release label on current component CR

Looking at this I don't think we necessarily require templating. 
Maybe I didn't understand the use case behind templating here, so correct me if I missed something.

Reply:
okay before we get into another discussion, ill just remove it :D we can just readd it if we need it later on